protocols/v2/roles-logic-sv2/src/utils.rs

```diff
@@ -276,48 +280,94 @@ impl TryFrom<CoinbaseOutput> for Script {
 /// [3] https://en.wikipedia.org/wiki/Negative_hypergeometric_distribution
 /// bdiff: 0x00000000ffff0000000000000000000000000000000000000000000000000000
 /// https://en.bitcoin.it/wiki/Difficulty#How_soon_might_I_expect_to_generate_a_block.3F
-pub fn hash_rate_to_target(h: f32, share_per_min: f32) -> U256<'static> {
+
+pub fn hash_rate_to_target(
```

Comment:
Why don't use f64?

Reply:
Having an f64 gives us extra precision, that is wasted on that we work with the hashrate value, which is an f32 according to specs. The same for all the other instances of f32 commented below.

Reply:
is not wasted cause you have less errors during the calculations. Using f64 everywhere is for sure a better solution. If we see that this and issues from a performance point of view we can always use an f32 but I don't thing that will be an issue. Replace f32 with f64 everywhere.

Reply:
There are few benefits having the variabile hashrate as f64, as it is an f32 by specs. They may be consistent when we measure the downstream hashrate based on the shares frequency, like here
https://github.com/GitGab19/stratum/blob/dev/roles/translator/src/downstream_sv1/diff_management.rs#L102
But they are wasted when it comes from the downstream (in an OpenExtendedMiningChannel, for example) as hashrate would be casted from f32 to f64, giving us the the illusion of having more precision but, in fact, it is an f32.


For being precise, the benefis having the extra precision on share_per_min is killed [here](https://github.com/GitGab19/stratum/blob/dev/protocols/v2/roles-logic-sv2/src/utils.rs#L301) when we perform  
`
    let h_times_s = hashrate * shares_occurrency_frequence;
`
in which the extra precision of shares_occurrency_frequence f64 gets wasted because hashrate is f32 when this value comes from the downstream.

Reply:
To use f64 in hash_rate_to_target and hash_rate_from_target requires also that the values passed to these functions are *true* f64, and not just a f32 cast. This means that also other parts of the code need to be changed, using f64 whenever possible. 
I think that this is appropriate to be done in a separate PR